src/BenchmarkDotNet/Code/CodeGenerator.cs

```diff
@@ -186,7 +187,15 @@ private static DeclarationsProvider GetDeclarationsProvider(Descriptor descripto
             return new NonVoidDeclarationsProvider(descriptor);
         }
 
+        private static string GetParamsInitializer(BenchmarkCase benchmarkCase)
+            => string.Join(
+                ", ",
+                benchmarkCase.Parameters.Items
+                    .Where(parameter => !parameter.IsArgument && !parameter.IsStatic)
+                    .Select(parameter => $"{parameter.Name} = default"));
```

Comment:
I think it would be better to assign the actual value, that way `init` will also work. You will also need to change the `GetParamsContent` implementation to skip instance values. A rename to `GetInstanceParams` and `GetStaticParams` would be in order as well.

Reply:
That's what I tried initially (see previous commit) but the problem is that dynamic parameters won't work since they require the instance.
Also `required` support is essential not to get nullable warnings, `init` is a less useful IMO.

Reply:
I can't see your initial commit. Are you referring to ParamsSource?

Reply:
I forced-pushed, it's in the PR history:
https://github.com/dotnet/BenchmarkDotNet/pull/2579/commits/469f034adf1722c4069109ac96efc125fc53333a
Yes, ParamsSource

Reply:
Sorry it's this one - https://github.com/dotnet/BenchmarkDotNet/commit/32495a953cb473a053bce18a944c738986b05818

Reply:
I even named the methods as you suggested :)

Reply:
I agree, `init` doesn't seem particularly useful for benchmarks. It just would've been nice to support it "for free" if it was possible.